standards-catalogue/Rakefile

```diff
@@ -17,6 +17,15 @@ def is_valid_schema(schema, file_to_validate)
   false
 end
 
+def is_valid_schema_string(schema, to_validate)
```

Comment:
is this needed?

Reply:
it's used here https://github.com/alphagov/data-standards-authority/blob/54ff77f185ff1a3d7ef572225934faea6d172a59/standards-catalogue/Rakefile#L57
But it's almost the same as the method above (is_valid_schema) except with that one you pass in a file and then load it. Could maybe get rid of that one and where it's used in the standards validation do:
```
desc "Validate Standards are schema compliant"
  task :standards do
    schema = schema("standards")
    errors = Dir.glob("../content/standards/*/*.md").map do |f|
      to_validate = YAML.load_file(f) 
      is_valid_schema_string(schema, to_validate)
    end
    raise unless errors.all?
  end
```
or something?

Reply:
I usually follow the "Rule of Three" (don't de-duplicate code until the third case, where there's more context about what's actually common between them, so we can then refactor more reasonably) 

So I'd say that this looks fine to be duplicated for now.

Not sure re naming of `is_valid_schema_string` though. Maybe we can rename this `is_valid_schema` and the other to `does_file_match_schema`?

Reply:
makes sense, yep that's much better naming